### ISSUE TITLE
Automatically use correct format_checker for versioned Validators

### ISSUE THREAD

**Issue by TiborVoelcker** [Julian/jsonschema#861]
_(Not sure if this is already a feature, and I just could not get it to work.)_

I want my validator automatically detected using the "$schema" keyword, to leave it customizable. But I would also need the formats to be validated. I know I could use `validate(instance, schema, format_checker=draft202012_format_checker)`. But I would like to just use the corresponding format_checker. Meaning, if my "$schema" keyword leads to using the `Draft202012Validator`, I would want to automatically use the `draft202012_format_checker`. If I use `Draft7Validator`, use the `draft7_format_checker`, etc.

My proposal would be to accept `format_checker=True`, to then use the corresponding Format Validator.

**Comment by Julian** [comment]
Thanks, sounds great, and appreciated!

**Comment by TiborVoelcker** [comment]
Hi! Sorry for lettings this sit for such a long time. I found some time a while ago, but was completely lost when looking through the code. 
Now I found time again, and was determined to succeed. I tried to get an overview, but I am still a bit overwhelmed. But if I assessed it correctly, the change needed for this is very minor. So I did make a PR #905.

BTW: I could not figure out how to assign you to the PR, or link this issue with the PR.
